WordPress/gutenberg
# Prepare image lightbox for private version of the Interactivity API in 6 4

## What?
Prepare the image lightbox to be compatible with the private version of the Interactivity API which will be included in 6.4 by:
* Changing the `view-interactivity.js` to `view.js`.
* Adding a couple of attributes to support the SSR.

## Why?
For the renaming, we decided to use `view.js` instead of `view-interactivity.js` for all interactive blocks.

Regarding the new attributes: the Interactivity API has its own logic to handle the SSR: [link](https://github.com/WordPress/gutenberg/tree/76b0165002690569cc693fefc6b3e94e7e09204a/lib/experimental/interactivity-api). However, this logic won't be included in WordPress 6.4, so we have to handle it manually until it can be used.

## How?
I just renamed the file and added a couple of aria attributes for the SSR.

It is important to note that I didn't add SSR support to:
* Styles of the lightbox: As the image is hidden by default, I believe we don't need this.
* Styles of the image button: The width and height of the image button are not correct because they are calculated via JS. Replicating that logic in the serves seems too complicated, and, as it doesn't create any layout shifts, I believe it is not a big issue.

## Testing Instructions
1. Disable the current Interactivity API SSR logic by commenting these lines: [link](https://github.com/WordPress/gutenberg/blob/6db51a6ee92db6c3324a875135656861bbd9e55d/lib/load.php#L113-L123). This way we can simulate how it will work in WordPress 6.4.
2. Add an image to a page and enable the "Expand on click" functionality.
3. Go to the frontend and Disable the JS through the Dev Tools (or view the page source).
4. Check that the proper HTML comes from the server without running JS.
